src/libvcs/sync/git.py

```diff
@@ -418,9 +418,9 @@ def update_repo(self, set_remotes: bool = False, *args: Any, **kwargs: Any) -> N
             # to be able to perform git pull --rebase
             if need_stash:
                 # If Git < 1.7.6, uses --quiet --all
-                git_stash_save_options = "--quiet"
+                git_stash_save_options = ["--quiet", "--include-untracked"]
```

Comment:
I have a proposal: 

Add a `stash_untracked_files: bool=False` to `update_repo()` and add this conditionally

The reason why is: This sounds like a good knob/option for future vcspull versions to allow configuring in the future

Reply:
P.S. if that is too tedious/ beyond scope I can also do it and we can merge this as-is (up to you!)

Reply:
Would that then be set to `True` in `sync.py` on the `vcspull` side?

Reply:
> Would that then be set to True in sync.py on the vcspull side?

Yes

Reply:
@jfpedroza On second thought I think this is safe enough for default behavior and merging as-is

What do you think? Any adverse side effects / consequences you can think of for users?

https://stackoverflow.com/a/6818797, https://stackoverflow.com/a/12681856

One of them mentions this:

> Warning, doing this will permanently delete your files if you have any directory/ entries in your gitignore file.*

I'm not fully sure what that means and how that will work at scale yet. I will look at this closer in the morning.

In regards to https://github.com/vcs-python/libvcs/issues/395#issue-1339940775, do you think this option is safest?

Reply:
I hadn't realized `git stash show` doesn't show untracked files.

Alternatively, we could do option 1. It leaves the untracked file in the working directory and performs the rebase anyway. If there are any conflicts with the incoming changes and the file, the rebase will be aborted.

![image](https://user-images.githubusercontent.com/12222295/192129100-7a0d9944-1cfc-467d-b6d0-2f59edf96c8f.png)

Reply:
@jfpedroza I haven't fully thought through this yet

> Alternatively, we could do option 1. It leaves the untracked file in the working directory and performs the rebase anyway. If there are any conflicts with the incoming changes and the file, the rebase will be aborted.

Can you make a second PR (third PR) with this as an examle?